component-ldap/src/main/java/com/synopsys/integration/alert/channel/ldap/model/LDAPGlobalConfigModel.java

```diff
@@ -0,0 +1,156 @@
+package com.synopsys.integration.alert.channel.ldap.model;
+
+import java.util.Optional;
+
+import org.apache.commons.lang3.StringUtils;
+
+import com.synopsys.integration.alert.api.common.model.Obfuscated;
+import com.synopsys.integration.alert.common.rest.AlertRestConstants;
+import com.synopsys.integration.alert.common.rest.model.ConfigWithMetadata;
+
+public class LDAPGlobalConfigModel extends ConfigWithMetadata implements Obfuscated<LDAPGlobalConfigModel> {
+    private Boolean enabled;
+    private String serverName;
+    private String managerDn;
+    private String managerPassword;
+    private Boolean isManagerPasswordSet;
+    private String authenticationType;
+    private String referral;
+    private String userSearchBase;
+    private String userSearchFilter;
+    private String userDnPatterns;
+    private String userAttributes;
+    private String groupSearchBase;
+    private String groupSearchFilter;
+    private String groupRoleAttribute;
+
+    public LDAPGlobalConfigModel() {
+        // For serialization
+    }
+
+    // Required
+    public LDAPGlobalConfigModel(String id) {
+        super(id, AlertRestConstants.DEFAULT_CONFIGURATION_NAME);
+    }
```

Comment:
1) We can drop the required comment as it could probably be a bit misleading between this and the constructor with ALL properties passed in.

2) We should expand this constructor to cover the required fields needed for LDAP and the getters for  non-required fields should be wrapped in Optional tags below